python/lsst/cp/pipe/defects.py

```diff
@@ -0,0 +1,691 @@
+
+            for datasetType in defectLists.keys():
+                exp = dataRef.get(datasetType)
+                defects = self.findHotAndColdPixels(exp, datasetType)
+
+                msg = "Found %s defects containing %s pixels in master %s"
+                self.log.info(msg, len(defects), self._nPixFromDefects(defects), datasetType)
+                defectLists[datasetType].append(defects)
+                if self.config.makePlots:
+                    sig = self.config.nSigmaDark if datasetType == 'dark' else self.config.nSigmaBright
+                    self._plot(dataRef, exp, visitList[0], sig, defects, datasetType)
+
+        elif self.config.mode == 'VISITS':
+            butler = dataRef.getButler()
+
+            if self.config.assertSameRun:
+                runs = self._getRunListFromVisits(butler, visitList)
+                if len(runs) != 1:
+                    raise RuntimeError(f"Got data from runs {runs} with assertSameRun==True")
+
+            for visit in visitList:
+                imageType = butler.queryMetadata('raw', self.config.imageTypeKey, dataId={'visit': visit})[0]
+                imageType = imageType.lower()
+                dataRef.dataId['visit'] = visit
+
+                if imageType == 'flat':  # note different isr tasks
+                    exp = self.isrForFlats.runDataRef(dataRef).exposure
+                    defects = self.findHotAndColdPixels(exp, imageType)
+                    defectLists['flat'].append(defects)
+
+                elif imageType == 'dark':
+                    exp = self.isrForDarks.runDataRef(dataRef).exposure
+                    defects = self.findHotAndColdPixels(exp, imageType)
+                    defectLists['dark'].append(defects)
+
+                else:
+                    raise RuntimeError(f"Failed on imageType {imageType}. Only flats and darks supported")
+
+                msg = "Found %s defects containing %s pixels in visit %s"
+                self.log.info(msg, len(defects), self._nPixFromDefects(defects), visit)
+
+                if self.config.makePlots:
+                    nSig = self.config.nSigmaBright if imageType == 'flat' else self.config.nSigmaDark
+                    self._plot(dataRef, exp, visit, nSig, defects, imageType)
+
+        msg = "Combining %s defect sets from darks for detector %s"
+        self.log.info(msg, len(defectLists['dark']), detNum)
+        mergedDefectsFromDarks = self._postProcessDefectSets(defectLists['dark'], exp.getDimensions(),
+                                                             self.config.combinationMode)
+        msg = "Combining %s defect sets from flats for detector %s"
+        self.log.info(msg, len(defectLists['flat']), detNum)
+        mergedDefectsFromFlats = self._postProcessDefectSets(defectLists['flat'], exp.getDimensions(),
+                                                             self.config.combinationMode)
+
+        self.log.info("Combining bright and dark defect sets")
```

Comment:
I think it's in keeping with all the other messages in the task. I will add the identifiers and keep it as an `INFO` level so that it's coherent. This task is not very verbose, nor is it long or slow, so it's nice to see what's happening.